api/host.go

```diff
@@ -1,153 +1,227 @@
 package api
 
 import (
-	//"fmt"
+	"errors"
+	"fmt"
 	"net/http"
 
+	"github.com/NebulousLabs/Sia/crypto"
 	"github.com/NebulousLabs/Sia/modules"
-	"github.com/NebulousLabs/Sia/types"
 
 	"github.com/julienschmidt/httprouter"
 )
 
+var (
+	// errStorageFolderNotFound is returned if a call is made looking for a
+	// storage folder which does not appear to exist within the storage
+	// manager.
+	errStorageFolderNotFound = errors.New("storage folder with the provided path could not be found")
+)
+
 type (
 	// HostGET contains the information that is returned after a GET request to
-	// /host.
+	// /host - a bunch of information about the status of the host.
 	HostGET struct {
-		AcceptingContracts bool               `json:"acceptingcontracts"`
-		MaxDuration        types.BlockHeight  `json:"maxduration"`
-		NetAddress         modules.NetAddress `json:"netaddress"`
-		RemainingStorage   uint64             `json:"remainingstorage"`
-		TotalStorage       uint64             `json:"totalstorage"`
-		UnlockHash         types.UnlockHash   `json:"unlockhash"`
-		WindowSize         types.BlockHeight  `json:"windowsize"`
-
-		Collateral             types.Currency `json:"collateral"`
-		ContractPrice          types.Currency `json:"contractprice"`
-		DownloadBandwidthPrice types.Currency `json:"downloadbandwidthprice"`
-		StoragePrice           types.Currency `json:"storageprice"`
-		UploadBandwidthPrice   types.Currency `json:"uploadbandwidthprice"`
-
-		AnticipatedRevenue types.Currency `json:"anticipatedrevenue"`
-		LostRevenue        types.Currency `json:"lostrevenue"`
-		NumContracts       uint64         `json:"numcontracts"`
-		Revenue            types.Currency `json:"revenue"`
-
-		RPCErrorCalls        uint64 `json:"rpcerrorcalls"`
-		RPCUnrecognizedCalls uint64 `json:"rpcunrecognizedcalls"`
-		RPCDownloadCalls     uint64 `json:"rpcdownloadcalls"`
-		RPCRenewCalls        uint64 `json:"rpcrenewcalls"`
-		RPCReviseCalls       uint64 `json:"rpcrevisecalls"`
-		RPCSettingsCalls     uint64 `json:"rpcsettingscalls"`
-		RPCUploadCalls       uint64 `json:"rpcuploadcalls"`
+		FinancialMetrics modules.HostFinancialMetrics `json:"financialmetrics"`
+		InternalSettings modules.HostInternalSettings `json:"internalsettings"`
+		NetworkMetrics   modules.HostNetworkMetrics   `json:"networkmetrics"`
+	}
+
+	// StorageGET contains the information that is returned after a GET request
+	// to /storage - a bunch of information about the status of storage
+	// management on the host.
+	StorageGET struct {
+		StorageFolderMetadata []modules.StorageFolderMetadata
 	}
 )
 
-// hostHandlerGET handles GET requests to the /host API endpoint.
+// folderIndex determines the index of the storage folder with the provided
+// path.
+func folderIndex(folderPath string, storageFolders []modules.StorageFolderMetadata) (int, error) {
+	for i, sf := range storageFolders {
+		if sf.Path == folderPath {
+			return i, nil
+		}
+	}
+	return -1, errStorageFolderNotFound
+}
```

Comment:
seeks like a hack to do this in the API server rather than in the host, but I can understand if you don't want to rewrite all the host methods right now